### DISCUSSION TITLE
Please give the AnimationTree more signals

### DISCUSSION THREAD

**Discussion by CodeLazier** [godotengine/godot-proposals#5287] [Engine Core]
The AnimationTree node does not provide any signal events related to animation playback. which results in many missing events for state animations, such as callbacks after animations play. At the moment I can only animate the player to add call functions to manually call back, which is very inelegant and boring. Please add more signal support to the AnimationTree, such as animation finished playing, animation starting to play, animation playback progress, state change, etc

**Comment by Mickeon** [comment]
I agree with most, but I'm not sure about "playback_progress". As the 3.x **Tween** proves, emitting a signal every frame turns out to be pretty "expensive" right now.

**Reply by CodeLazier** [reply]
Indeed, if this affects running performance, I'm in favor.after all, the animationplayer node does not have this signal.
